sumo/plotting/dos_plotter.py

```diff
@@ -265,17 +272,30 @@ def get_plot(self, subplot=False, width=None, height=None, xmin=-6.,
         energies = plot_data['energies'][mask]
         fig = plt.gcf()
         lines = plot_data['lines']
-        spins = [Spin.up] if len(lines[0][0]['dens']) == 1 else \
-            [Spin.up, Spin.down]
+        if len(lines[0][0]['dens']) == 1:
+            spins = [Spin.up]
+        elif spin == 'up':
+            spins = [Spin.up]
+        elif spin == 'down':
+            spins = [Spin.down]
+        else:
+            spins = [Spin.up, Spin.down]
 
         for i, line_set in enumerate(plot_data['lines']):
             if subplot:
                 ax = fig.axes[i]
             else:
                 ax = plt.gca()
 
+            ax.set_ylim(plot_data['ymin'], plot_data['ymax'])
```

Comment:
Why is this happening inside a loop over lines when it only depends on the overall plot data? Shouldn't need to call this more than once.